scripts/getHymns.py

```diff
@@ -1,124 +1,245 @@
+    """
+    if name_to_check not in list_of_names:
+        return name_to_check
+    counter = 1
+    if len(name_to_check.split(separator)) > 1:
+        counter = int(name_to_check.split('_')[1]) + 1
+    return check_name(f"{name_to_check.split(separator)[0]}{separator}{counter}", list_of_names)
+
+def download_catalog(url, save_hinario=False, save_hino=False):
+    """ Scarica l'intero catalogo. """
+    req = requests.get(url, timeout=10)
+    soup = BeautifulSoup(req.content, 'html.parser')
+
+    catalog = {
+        'title': soup.select('.page_breadcrumbs h2')[0].get_text().strip(),
+        'hinarios': []
+    }
+
+    hinarios_raw = soup.select('h4 a')
+    for hinario_raw in hinarios_raw:
+        hinario_url = hinario_raw.attrs['href']
+        hinario = download_hinario(hinario_url, save_hino=save_hino)
+        if save_hinario:
+            hinario_json = json.dumps(hinario, indent=4)
+            filename = f"./{SAVE_DIR}/{hinario['person']} - {hinario['title']}.json"
+            with open(filename, 'w', encoding='utf-8') as outfile:
+                outfile.write(hinario_json)
+        catalog['hinarios'].append(hinario)
+    return catalog
+
+def download_hinario(url, save_hino=False):
+    """ Scarica il singolo hinario con tutti gli hino a lui relativi. """
+    logging.debug('Scarico hinario: %s', url)
+
+    req = requests.get(url, timeout=10)
+    soup = BeautifulSoup(req.content, 'html.parser')
+
+    hinario = {
+        'person': soup.select('.breadcrumb li:nth-last-of-type(2) a')[0].get_text().strip(),
+        'title': soup.select('.page_breadcrumbs h2')[0].find(string=True, recursive=False).get_text().strip(),
+        'hinos': []
+    }
+
+    logging.info('+ %s', hinario['title'])
+
+    hinos_raw = soup.select('.hymn-list-name a')
+    for hino_raw in hinos_raw:
+        hino_url = hino_raw.attrs['href']
+        hino = download_hino(hino_url, hinario['title'])
+        if save_hino:
+            hino_json = json.dumps(hino, indent=4)
+            filename = f"./{SAVE_DIR}/{hino['person']} - {hino['hinario']} - {hino['index']}. {hino['title']}.json"
+            with open(filename, 'w', encoding='utf-8') as outfile:
+                outfile.write(hino_json)
+        # Essendo già presenti nei dati dell'hinario, chi lo ha ricevuto
+        # e il titolo sono informazioni ridondanti.
+        del hino['person']
+        del hino['hinario']
+        hinario['hinos'].append(hino)
+    return hinario
+
+def download_hino(url, hinario=''):
+    """ Scarica un singolo hino. """
+    logging.debug('Scarico hino: %s', url)
+
+    req = requests.get(url, timeout=10)
+    soup = BeautifulSoup(req.content, 'html.parser')
+
+    index = soup.select('.hinario-breadcrumb-hinario a')[0].get_text().strip().split('#')[1]
+
+    # Se alla funzione non viene passato il titolo dell'hinario, significa che
+    # si sta scaricando il singolo hino, e perciò verranno presi per validi
+    # il titolo e l'indice nella barra principale.
```

Comment:
titolo e l'indice? non ho trovato l'indice

Reply:
Con indice intendo il numero dell'hino nella sequenza interna all’hinario. Lo stesso hino può essere presente in più di un hinario con indice diverso.